app/policies/portfolio_item_policy.rb

```diff
@@ -45,11 +45,19 @@ def can_read_and_update_destination?(destination_id)
 
   class Scope < Scope
     def resolve
-      if catalog_administrator?
+      access_scopes = @user_context.access.scopes(scope.table_name, 'read')
+
+      if access_scopes.include?('admin')
         scope.all
-      else
+      elsif access_scopes.include?('group')
```

Comment:
@eclarizio 
Here I think we should be getting values from the params to get the :portfolio_id
https://github.com/RedHatInsights/catalog-api/blob/89666f3032cc26f5463d1845a5d38b347cabbaba/app/controllers/api/v1x0/portfolio_items_controller.rb#L7
Otherwise we will be passing back portfolio_items from other portfolios.

Reply:
Also it seems that whole check in the portfolio_items_controller#index should get moved here

Reply:
I don't think so, line 8 does:
`Portfolio.find(params.require(:portfolio_id)).portfolio_items`, so our scope is already within that portfolio. Then we're doing a `.where(:portfolio_id => ids)` in the scope, it just makes sure that the ACEs for that ID are in the table.

So, ultimately the query built I think ends up being:

`Portfolio.find(params.require(:portfolio_id)).portfolio_items.where(:portfolio_id => ids)`, and that last `.where` part is kind of redundant, but it does end up potentially returning an empty set if the portfolio_id that got passed into the parameters doesn't have an associated ACE.